File: src/main/scala/com/github/mauricio/postgresql/messages/backend/DataRowMessage.scala
package com.github.mauricio.postgresql.messages.backend

import org.jboss.netty.buffer.ChannelBuffer

/**
 * User: mauricio
 * Date: 3/31/13
 * Time: 1:10 AM
 */
case class DataRowMessage( val values : Array[ChannelBuffer] ) extends Message( Message.DataRow )

File: postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/messages/backend/DataRowMessage.kt
package com.github.jasync.sql.db.postgresql.messages.backend

import io.netty.buffer.ByteBuf
import java.util.*

data class DataRowMessage(val values: Array<ByteBuf?>) : ServerMessage(ServerMessage.DataRow) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as DataRowMessage

        if (!Arrays.equals(values, other.values)) return false

        return true
    }

    override fun hashCode(): Int {
        return Arrays.hashCode(values)
    }
}
